docs/rfc/0003-collaborative-objects.md

```diff
@@ -0,0 +1,649 @@
+  conflicting changes, we could expose the merge points of collaborative
+  objects to the user to resolve.
+
+The latter here is undesirable for a lot of reasons but most relevant is that
+it conflicts with extensibility and interoperability. in order to provide a good 
+UX tools would need to provide a UI for users to do a three way merge of the 
+data they are working on. This contradicts the interoperability goal.
+
+This leaves us with CRDTs. For the purposes of this RFC I am going to assume
+that we will use the [Automerge](https://github.com/automerge/automerge) CRDT
+implementation. Automerge is capable of merging arbitrary JSON objects (the
+data model is actually richer than JSON, with types for byte arrays, various
+precision integers and floats etc.). An alternative approach would be to write
+a custom CRDT for each data type we want to replicate, see [##Alternative
+Approaches] for a discussion of this design.
+
+## Example: Issues
+
+To motivate and contextualise this proposal we introduce a running example
+which is dear to many peoples hearts; issues. We imagine this from the
+perspective of the application developer. For expositional purposes we assume
+that the application communicates with a radicle implementation via an HTTP
+RPC. However, the HTTP RPC is not a proposal of this RFC, the proposed API will
+be specified in terms of `librad` later in this document.
+
+The first thing the developer must do is decide on the schema of their data and
+represent it as a JSON schema. We use this simple schema:
+
+```json
+{
+  "type": "object",
+  "rad_signed_by": {
+    "fields": ["title", "description", "author"],
+    "keys": ["<authors URN>"]
+  },
+  "properties": {
+    "id": {
+      "type": "string",
+      "frozen": true
+    },
+    "title": {
+        "type": "string",
+        "automerge_type": "text"
+    },
+    "description": {
+        "type": "string",
+        "automerge_type": "text"
+    },
+    "author": {
+        "description": "The radicle ID of the author of the issue",
+        "const": "<the authors URN>",
+        "frozen": true
+    },
+    "comments": {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "rad_signed_by": {
+              "properties": ["text", "author"],
+              "keys": ["<commenters URN>"]
+            },
+            "properties": {
+                "text": {"type": "string"},
+                "author": {
+                    "type": "string",
+                    "description": "Radicle URN of the author of the comment",
+                    "automerge_type": "text"
+                }
+            }
+        }
+    }
+  }
+}
+```
+
+An issue consists of a title, description, and author along with the author's
+signature; followed by a list of comments, each of which is signed by it's
+respective author. This is an extremely simplified model. Note the presence of
+the `automerge_type`, `frozen`, and `rad_signed_by`, and `rad_urn` keys. 
```

Comment:
It would be good to link to the [Schema Extensions](https://github.com/radicle-dev/radicle-link/pull/662/files#diff-a302abf8a4acae0a3e0b65f64f9079fadd3ae105bff1978cf2736f498ba92be6R431) section here.